pynsee/utils/save_df.py

```diff
@@ -0,0 +1,126 @@
+
+                insee_date_time_now = kwargs.get("insee_date_test", False)
+                
+                if insee_date_time_now is not False:                    
+                                        
+                    insee_date_time_now = datetime.datetime.strptime(
+                        insee_date_time_now, "%Y-%m-%d %H:%M:%S.%f"
+                    )
+                else:
+                    insee_date_time_now = datetime.datetime.now()
+
+                day_lapse = (insee_date_time_now - file_date_last_modif).days
+
+                if day_lapse_max is not None:
+                    if day_lapse > day_lapse_max:
+                        update = True                    
+            
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=ShapelyDeprecationWarning)
+
+                if (not os.path.exists(file_name)) or update:   
+
+                    df = func(*args, **kwargs)                    
+                    try:
+                        if parquet:
+                            df.to_parquet(file_name)
+                        else:
+                            df.to_csv(file_name, index=False)
+                            
+                    except Exception as e:
+                        warnings.warn(str(e))
+                        warnings.warn(f'Error, file not saved:\n{file_name}\n{df}')
+                        warnings.warn('\n')
+                    
+                    df = obj(df) 
+                    
+                    if not silent:
+                        logger.info(f"Data saved:\n{file_name}")
+
+                else:
+                    try:                        
+                        if parquet:
+                            df = pd.read_parquet(file_name)
+                        else:
+                            df = pd.read_pickle(file_name)
+                            
+                        if 'Unnamed: 0' in df.columns:
+                            del df['Unnamed: 0']
+                            
+                    except Exception as e:
+                        warnings.warn(str(e))
+                                                
+                        kwargs2 = kwargs
+                        kwargs2['update'] = True
+                        
+                        warnings.warn('!!! Unable to load data, function retriggered !!!')
+                        
+                        df = func(*args, **kwargs2)
+                        df = obj(df)   
+                    else:
+                        if not silent:
+                            
+                            mdate = insee_date_time_now - datetime.timedelta(days=day_lapse)
+                            
+                            _warning_cached_data(file_name,
+                                                 mdate= mdate,
+                                                 day_lapse=day_lapse)
+                        df = obj(df)            
```

Comment:
remove

Reply:
4cf3d511b1bd7942b9075ccc463d4664fdd5ff5c should tackle the raised issue